lib/open_api/serializer.rb

```diff
@@ -1,14 +1,42 @@
 module OpenApi
   module Serializer
     def as_json(arg = {})
-      return super unless kind_of?(ApplicationRecord)
+      previous = super
+      encrypted_columns_set = (self.class.try(:encrypted_columns) || []).to_set
+      encryption_filtered = previous.except(*encrypted_columns_set)
+      version = version_from_path(arg)
 
-      super.tap do |hash|
-        hash.each_key do |attr_key|
-          hash[attr_key] = hash[attr_key].iso8601 if hash[attr_key].kind_of?(Time)
-          hash[attr_key] = hash[attr_key].to_s if attr_key.ends_with?("_id") || attr_key == "id"
-        end
+      return encryption_filtered unless arg.key?(:prefixes) || version
+
+      schema = Api::Docs[version].definitions[self.class.name]
+
+      filter(encryption_filtered.slice(*schema["properties"].keys), schema, encrypted_columns_set)
+    end
+
+    private
+
+    def version_from_path(arg)
+      url = ManageIQ::API::Common::Request.current.instance_variable_get(:@original_url) || nil
+      base_path = url ? URI.parse(url).path : arg[:prefixes].first
+      api_version_from_prefix(base_path)
+    end
+
+    def filter(filtered, schema, encrypted_columns_set)
```

Comment:
I'm not a big fan of the `filtered` var name, took me a bit to get that that is what's going to get returned. Maybe change it to something like `output` or `fields` etc.